src/js/components/Typeahead.js

```diff
@@ -0,0 +1,73 @@
+import InnerTypeahead from 'mesosphere-react-typeahead';
+/* eslint-disable no-unused-vars */
+import React, {PropTypes} from 'react';
+/* eslint-enable no-unused-vars */
+
+import FilterInputText from './FilterInputText';
+
+class Typeahead extends FilterInputText {
+  handleInputChange(input) {
+    this.props.handleFilterChange(input);
+  }
+
+  componentWillUpdate() {
+    if (this.inputField == null && this.typeahead) {
+      if (this.typeahead.refs.inner) {
+        if (this.typeahead.refs.inner.inputComponent) {
```

Comment:
Instead of the nested `if`s, lets just make them all a single `if` chained by `&&`